### PR TITLE
wallet: GetEffectiveBalance

### PR DESCRIPTION
**Pull request by willcl-ark**
This is a rough draft of what a `GetEffectiveBalance` function on the wallet might look like, along with RPC call and wallet interface.
Related to #15767

Still todo and open questions:

- RPCHelp wording is just a rough draft and needs amending
- Should the result contain additional info such as: fee rate used, num utxos, num excluded utxos etc. (this would better satisfy #15767 too)
- Should confirm target be configurable? Currently fixed to a 3 blocks.
  - In this case the response should likely include the confirm target used to generate the effective value.
- Should this be added to QT wallet? Interface is present. I think it would be nice but don't know much about QT.
- ~Required adding a lot of headers to `receive.cpp` which seems sub-optimal. Perhaps the function belongs somewhere else...~
- Doesn't add a dummy recipient to the tx, so fee calculation is slightly off
- ~Doesn't respect `avoid_reuse`~
- Doesn't respect `watch_only`
- RBF just uses wallet default

### CODE REVIEW

File: src/wallet/rpc/coins.cpp [hunk 1]
```diff
@@ -213,6 +213,59 @@ RPCHelpMan getbalance()
     };
 }
 
+RPCHelpMan geteffectivebalance()
+{
+    return RPCHelpMan{"geteffectivebalance",
+                "\nReturns the total effective balance.\n"
+                "The effective balance is what the wallet considers currently spendable at a\n"
+                "confirmation target of 3 blocks at current feerate.\n",
```
Comment (luke-jr) [inline src/wallet/rpc/coins.cpp:221]: Why? This seems quite arbitrary.

**Comment by Xekyo** [comment]
Concept ACK

I think it's useful to be able to tell users how much they could actually spend. A three-block target or six-block target seems like a reasonable default, but I'd expect this RPC call to be more useful if it could also take a custom feerate.

**Comment by aureleoules** [comment]
Concept ACK

I agree that there should be a configurable fee rate and/or configurable block target.
How about adding an RPC arg `effective` to `getbalance` instead of creating a separate RPC?

**Comment by willcl-ark** [comment]
Going to close this as there are other, cleaner and better-maintained version(s) of it being worked on now.

**Pull request closed without merging**